### ISSUE TITLE
Logo for Extraterm

### ISSUE THREAD

**Issue by sedwards2009** [g-harel/open-source-logos#4]
## Extraterm
https://github.com/sedwards2009/extraterm

### Background
It is a terminal emulator, but the idea with this project is to take the traditional terminal emulator and modernise it with new and original features which make sense in the 21st century. For example, the ability to reuse previous command output directly and to edit it, is a feature which no traditional terminal has. I'm trying to break out of the old mold of a terminal application just being an emulation of an old piece of hardware which no one has seen in over 20 years.

### Description
One way of explaining what Extraterm is, is to say its the swiss army knife or leatherman of terminal emulators. As it grows it will be getting lots of different and useful tools added. I see it as a playful project and not super serious. It is a bit of an experiment too which may end up being a bit "over the top". (i.e. "the swiss army chainsaw of terminal emulators"). I don't mind "over the top". ;-)

I think the swiss army knife, leatherman, multi-tool, power-tool angle makes the most sense.

If it could also communicate a link to the concept "terminal", that would be nice to have, but it certainly isn't a requirement. (Most logos for terminal emulators tend to be boring rectangles with a >_ prompt inside. I would prefer an interesting logo which doesn't say "terminal" over a boring one which does. If some kind of reference to terminals doesn't fit with the design, then drop it.)

### Time Constraints
No real time constraints.

**Comment by g-harel** [comment]
Could I get your thoughts on this direction ?

I had a lot of trouble trying to incorporate the swiss army knife + terminal + chainsaw, so I ended up going for a simpler look.


![image](https://user-images.githubusercontent.com/9319710/26911028-be67f3ec-4bd8-11e7-9844-5fc47ecd54e9.png)

**Comment by sedwards2009** [comment]
I think it is a really good start. I can imagine that it is hard to combine the shopping list of concepts I gave you. ;-)

The logo so far communicates the 2 most important ideas about the project: terminal + many (little) tools, and that is the most important thing.

please continue.

**Comment by g-harel** [comment]
Can I get an idea of what direction you would like me to go from here, please be as specific as you can. For example, I was thinking of a blade/tool that is sticking out a bit from the side, but I was worried it might become too busy.
I also intend on adding a type underneath (if you want that) and if you have any ideas for that, let me know.

**Comment by sedwards2009** [comment]
Yes, I was thinking of continuing this direction with the knife and adding blades/tools. As far as being too busy is concerned, I can see that if you add a full length blade it won't leave much room over for anything else, not without it all overlapping. An idea might be to draw 2 of the smaller half length tools, one at each end. At least then they won't collide with each other. (That would give the whole form an almost triangular shape which might not be a bad thing.)

As for text, that sounds fine, but I'm out of ideas regarding how best to do it.

**Comment by g-harel** [comment]
![image](https://user-images.githubusercontent.com/9319710/28147488-d858c862-674e-11e7-8a41-1d0be481e67d.png)

**Comment by sedwards2009** [comment]
It's looking good! 

Maybe the blade could be a bit chunkier/thicker though.

I've been thinking and doing some googling about the design here. It turns out a certain manufacturer of "portable cutlery" has basically a trademark on the colour red when used on a tool handle, and they aren't shy to sue anyone who gets near it. So in the interests of not getting sued I think it is a good idea to change the colour. Green is the best candidate, also because it can act as a reference back to the days when most (hardware) terminals were monochrome green screens. https://www.google.nl/search?q=green+screen+terminal&num=20&source=lnms&tbm=isch&sa=X&ved=0ahUKEwjXmNWy_IbVAhVHJlAKHZqADkoQ_AUICigB&biw=1023&bih=1327

(If you think some kind of texture is worth experimenting with, try this http://imgur.com/BDNxJeS )

With those changes I think this would have to be very close to done. The only other request I have is for a small logo. I'm thinking a disc with the same highlights/shadows as above and with the '>_' in the middle. Basically as though you took the two ends of the tool body and put them together to form a circle.

thanks for your continuing work!

**Comment by g-harel** [comment]
If I'm understanding this correctly it would be a black handle with green writing ?

**Comment by sedwards2009** [comment]
Green handle, white text.

Would black or dark grey even work?

**Comment by g-harel** [comment]
I think it could work, I'll make both and you can see which one is best

**Comment by g-harel** [comment]
![image](https://user-images.githubusercontent.com/9319710/28236160-838d16f4-68ec-11e7-97f5-a47bc2c56531.png)

**Comment by sedwards2009** [comment]
I'm leaning towards the white text on green. Black handle looks a bit too serious/drab.

I was curious about green text on black so I hacked this up in GIMP.

![extraterm_logo_green_on_black](https://user-images.githubusercontent.com/6926644/28238114-fd6be10c-694c-11e7-8154-b834972d9b5f.png)

I'm not super impressed with the result and most of the logo would disappear if it were ever put on a black background. So I think I'll have to go with white text on black for the main logo.

Then I was thinking about black backgrounds, and I quickly did this in GIMP:

![logo_neon_green](https://user-images.githubusercontent.com/6926644/28238152-c6865446-694d-11e7-8696-1ee0edc1570b.png)

It looks retro as all hell, but I have to admit that I think it looks cool and I can think of situations where I have a black background and want a logo. For example, I could use it in the little Tip-of-the-Day things which appear in the terminal area which usually has a straight black background. 

In summary to finish this all off what I need is:

* Main logo - (White text on green handle, suitable for white background) You've kind of done this already.
* Secondary version of logo - (Outline/edges in green, suitable for black background). Can you do a proper version of this please?
* Icon - (Like the main logo, green disc with '>_' text and highlights)

Renders in different sizes are needed, but having the source material in SVG format would be better, as long as Inkscape can understand it and render it correctly. (Are fonts likely to be a problem?)

Thanks for your time and effort. I hope I'm not being too greedy here. When I put all this stuff together at my end, how should I credit and/or link back to you and your work?

**Comment by g-harel** [comment]
Yeah I tried the green text on black, but it looked terrible so I didn't post it haha.

Outlines shouldn't be a problem.

**Comment by g-harel** [comment]
I've added the three files in this directory

https://github.com/g-harel/open-source-logos/tree/master/projects/Extraterm/images

👍

**Comment by sedwards2009** [comment]
Thanks man, they look great! 🏅 

A small technical note: Inkscape had some trouble with the files. The opacity of two parts was busted. It turns out that whatever software you used for the SVGs, it expressed `0.05` as `5.000000000e-02` which Inkscape doesn't understand. I corrected this problem by hand and now it is fine.

Regarding the small icon, the loop in the top right doesn't add much and it is going to be unrecognisable at the sizes I'm going to need to use the icon in (i.e. small). I can just remove that part myself in Inscape. Other than that it looks great too.

Thanks for the outline version also.

I'm looking forward to updating the `readme.md` and the About tab in the app with the new logo's and icons. It will push the initial impressions of this project up big notch or two. At some time in the near future I'll be putting up a simple website for the project too which of course will feature your work on the front page.

How can I best credit and link back to you in the docs and About tab? I assume you are fine with logos and svgs being put under the MIT license.

thank you again.

**Comment by g-harel** [comment]
I used Illustrator to create them, sorry about the opacity stuff, I've never tried out Inkscape.

You can link back to my profile if you want to, but no pressure.

Yep, use whatever license you like.

I'm glad you like it, good luck with the app !
